### PR TITLE
:seedling: Backport v1alpha2 to release-1.0 branch

### PR DESCRIPTION
**Pull request by nprokopic**
Towards giantswarm/roadmap#799

We want to have our changes in `release-x.y` branches, because in upstream those branches are used for releases.

Adds Giant Swarm changes to `release-1.0` branch:
- CircleCI config
- Backporting v1alpha2 types

### CODE REVIEW

**Review by fiunchinho — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -296,9 +296,11 @@ generate-go-core: $(CONTROLLER_GEN)
 
 .PHONY: generate-go-conversions-core
 generate-go-conversions-core: $(CONVERSION_GEN)
+	$(MAKE) clean-generated-conversions SRC_DIRS="./api/v1alpha2"
```
Comment (tfussell) [inline Makefile:299]: No experimental APIs in v1alpha2?

**Review by tfussell — APPROVED** [review]

**Comment by tfussell** [comment]
Are you going to port it to `release-1.1` branch too?

**Review by calvix — APPROVED** [review]

**Comment by nprokopic** [comment]
> Are you going to port it to `release-1.1` branch too?

Yeah, I think that we will have to keep porting it to every release-x.y branch, because those are the stable branches that the releases are coming from.

**Comment by fiunchinho** [comment]
Tested on `ginger` and `godsmack`, no issues found.

**Pull request merged**